internal/kafka/internal/clusters/ocm_provider.go

```diff
@@ -534,3 +535,22 @@ func (o *OCMProvider) GetClusterResourceQuotaCosts() ([]types.QuotaCost, error)
 	}
 	return quotaCostList, nil
 }
+
+func (o *OCMProvider) CheckIfOrganizationIsTheClusterOwner(externalOrganizationID, clusterID, clusterExternalID string) error {
+	organizationID, err := o.ocmClient.GetOrganisationIdFromExternalId(externalOrganizationID)
+	if err != nil || shared.StringEmpty(organizationID) {
+		return svcErrors.NewWithCause(svcErrors.ErrorGeneral, err, "failed to retrieve organization details")
+	}
+
+	search := fmt.Sprintf("cluster_id = '%s' AND external_cluster_id = '%s' AND organization_id = '%s'", clusterID, clusterExternalID, organizationID)
+	subcriptions, err := o.ocmClient.GetSubscriptions(0, 1, search)
```

Comment:
Seeing `o.ocmClient.GetSubscriptions(0, 1, search)` is not a very good interface: What is 0? and 1?

I think we can create a data type for page and size. For example `PagingOptions`